### PR TITLE
chore: add support for Matomo and Hotjar environment variables in docker compose files

### PR DESCRIPTION
**Pull request by e0**
**Related issue(s) and PR(s)**  
This PR closes https://github.com/MetabolicAtlas/private-issues/issues/122.

**Type of change**  

- [x] Other
 
**List of changes made**  

This adapts the usage of environment variables to be more consistent with the recent rework of handling other environment specific changes.

Previously, the now removed `frontend/.env.production` file was used for both the dev and prod servers. Now it is possible to set different values for each remote server by editing the `env-*.env` files. The specific Matomo ID for each environment can be found in the related issue.

**Testing**  

- Instructions on how to test

It is most straightforward to test this by deploying to the dev server.
1. Edit your `env-dev.dev` file according to the updated `env-local.env.sample` file. Only add the `VUE_APP_MATOMOID` as `VUE_APP_HOTJAR` should only be added for the prod server.
2. Deploy to the dev server: `source proj.sh` followed by `deploy-stack dev`.
3. After the deployment, visit the dev server in a browser and open the Network tab in the developer tools pane.
4. Verify that the correct Matomo ID is set in the `matomo.php?...` request as shown in the screenshot. It helps to enter `matomo` in the Filter field to help find the request.

image

**Further comments**  

I'm not sure if the Matomo id is needed for the local environment. If not, we could discard the changes for the `docker-compose-local.yml` file. On the other hand, perhaps it is ok to leave it in as we can simplify omit setting the variable by not adding it the `env-local.env` file.

**Checklist**  

- [x] My code follows the [NBIS style guidelines](https://github.com/NBISweden/development-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked so that the same data as before is returned by the API

### CODE REVIEW

**Comment by mihai-sysbio** [comment]
> 1. Only add the `VUE_APP_MATOMOID` as `VUE_APP_HOTJAR` should only be added for the prod server.

This makes a lot of sense to me, but it also generates this new warning:
```
WARN[0000] The "VUE_APP_HOTJAR" variable is not set. Defaulting to a blank string.
```

I'm not bothered by it, but remember us trying to have as few of these warnings as possible.

**Review by mihai-sysbio — APPROVED** [review]
Really neat way of dropping the (now-confusing) `frontend/.env`! Tested on `dev`, works great with the instructions. Just a side-note: if your browser is set to send the "Do Not Track" header, then the request will not be made.

File: env-local.env.sample [hunk 1]
```diff
@@ -50,3 +50,11 @@ PORT_NGINX=80
...
 PORT_NEO4J_1=7474
 PORT_NEO4J_2=7687
+
+## Matomo instance ID, used in frontend.
+## This should have different values for dev and prod servers.
+VUE_APP_MATOMOID=14
+
+## Hotjar script, used in frontend. 
+## This should only be added for the prod server.
+VUE_APP_HOTJAR=<script>if(navigator.doNotTrack !== '1') (function(h,o,t,j,a,r){h.hj=h.hj||function(){(h.hj.q=h.hj.q||[]).push(arguments)};h._hjSettings={hjid:1356321,hjsv:6};a=o.getElementsByTagName('head')[0];r=o.createElement('script');r.async=1;r.src=t+h._hjSettings.hjid+j+h._hjSettings.hjsv;a.appendChild(r);})(window,document,'https://static.hotjar.com/c/hotjar-','.js?sv=');</script>
```
[the remark this answers is not in the document]
Comment (e0) [reply env-local.env.sample:60]: Do you mean just for `VUE_APP_MATOMOID` or for `VUE_APP_HOTJAR` as well?

File: docker-compose-local.yml [hunk 2]
```diff
@@ -12,6 +12,8 @@ services:
       - ./nginx/svgs:/svgs
 
   frontend:
+    environment:
+      - VUE_APP_MATOMOID=${VUE_APP_MATOMOID}
```
[the remark this answers is not in the document]
Comment (e0) [reply docker-compose-local.yml:16]: That sounds reasonable. We just have to keep in mind that if we ever want to test anything related to Matomo we would need to deploy it to the dev server first.

**Review by nanjiangshu — APPROVED** [review]
Works great as I tested @e0.

I agree with @mihai-sysbio that the value for adding the MATONO ID in the local deployment is very little (maybe for testing, but we can surely use the `dev` server as well)

It seems the there is a predefined matono id since when I set the id to `1014` the status code is `400` for matomo.php.

When I tried to inspect the activity of matomo for https://metabolicatlas.org/, I could not find any record, I guess this is expected?

File: env-local.env.sample (hunk 1 above)
[the remark this answers is not in the document]
Comment (mihai-sysbio) [reply env-local.env.sample:60]: Both of them.

**Comment by mihai-sysbio** [comment]
> It seems the there is a predefined matono id since when I set the id to `1014` the status code is `400` for matomo.php.
> 
> When I tried to inspect the activity of matomo for https://metabolicatlas.org/, I could not find any record, I guess this is expected?

The id has to be first defined in Matomo, so `1014` doesn't work because there is no such tracking id set up.

It might be the case that you have instructed your browser manually (or automatically with Safari) to send the `Do Not Track` header, which disables the Matomo functionality, i.e. there will be no request made.

**Comment by nanjiangshu** [comment]
> > It seems the there is a predefined matono id since when I set the id to `1014` the status code is `400` for matomo.php.
> > When I tried to inspect the activity of matomo for https://metabolicatlas.org/, I could not find any record, I guess this is expected?
> 
> The id has to be first defined in Matomo, so `1014` doesn't work because there is no such tracking id set up.
> 
> It might be the case that you have instructed your browser manually (or automatically with Safari) to send the `Do Not Track` header, which disables the Matomo functionality, i.e. there will be no request made.

I don't think so. I'm using Firefox and the tracking configuration is `only when Firefox is set to block known trackers`.
For https://dev-metatlas.csbi.chalmers.se/, I see requests to `https://csbi.chalmers.se//matomo.php?action_name`, but for 
https://metabolicatlas.org/ I see requests to `https://csbi.chalmers.se//piwik.php?` instead. Maybe this is how it is set up?

**Pull request merged**